tests/org.jboss.tools.jsf.ui.test/src/org/jboss/tools/jsf/ui/test/project/RunJSFProjectTest.java

```diff
@@ -114,15 +164,27 @@ private void checkDeployedProject() {
 	}
 
 	private void deployProject() {
+		new WaitUntil(new JobIsRunning(), TimePeriod.MEDIUM, false);
```

Comment:
What do you wait for?

Reply:
Have to wait for server adapter to get created, sometimes I got error here saying, there is no server adapter... but in stack trace you can see it. This seemed to solve the problem.

Reply:
Perhaps this should be fixed in server requirement or here: https://github.com/eclipse/reddeer/blob/820e8f1891451870c0569173864b76f0286bd2c4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/wizard/NewServerWizard.java#L38

@rawagner any suggestions?

Reply:
We could add a wait which checks the TreeItem presence in the ServersView. Issue is here https://github.com/eclipse/reddeer/issues/1899